packages/at_secondary_server/lib/src/verb/handler/pkam_verb_handler.dart

```diff
@@ -38,58 +39,122 @@ class PkamVerbHandler extends AbstractVerbHandler {
-          '$enrollId.$newEnrollmentKeyPattern.$enrollManageNamespace$atSign';
-      var enrollData = await keyStore.get(key);
-      if (enrollData != null) {
-        final atData = enrollData.data;
-        logger.finer('enrollData: $atData');
-        final enrollDataStoreValue =
-            EnrollDataStoreValue.fromJson(jsonDecode(atData));
-        if (enrollDataStoreValue.approval != null &&
-            enrollDataStoreValue.approval!.state ==
-                EnrollStatus.approved.name) {
-          publicKey =
-              EnrollDataStoreValue.fromJson(jsonDecode(atData)).apkamPublicKey;
-          pkamAuthType = AuthType.apkam;
-        } else {
-          response.data = 'failure';
-          response.isError = true;
-          response.errorMessage = 'enrollment id: $enrollId is not approved';
-          throw UnAuthenticatedException(response.errorMessage);
-        }
+      pkamAuthType = AuthType.apkam;
+      ApkamVerificationResult apkamResult =
+          await handleApkamVerification(enrollId, atSign);
+      if (apkamResult.response.isError) {
+        response.isError = apkamResult.response.isError;
+        response.errorCode = apkamResult.response.errorCode;
+        response.errorMessage = apkamResult.response.errorMessage;
+        return;
       }
+      publicKey = apkamResult.publicKey;
     } else {
+      pkamAuthType = AuthType.pkamLegacy;
       var publicKeyData = await keyStore.get(AT_PKAM_PUBLIC_KEY);
       publicKey = publicKeyData.data;
     }
 
-    // If there is no public key in the keystore then throw an exception
     if (publicKey == null || publicKey.isEmpty) {
-      response.data = 'failure';
-      response.isError = true;
-      response.errorMessage = 'pkam publickey not found';
       throw UnAuthenticatedException('pkam publickey not found');
     }
-    var isValidSignature = false;
+    var isValidSignature =
+        await _validateSignature(verbParams, sessionID, atSign, publicKey);
+    // authenticate if signature is valid
+    if (isValidSignature) {
+      atConnectionMetadata.isAuthenticated = true;
+      atConnectionMetadata.authType = pkamAuthType;
+      atConnectionMetadata.enrollmentId = enrollId;
+      response.data = 'success';
+    } else {
+      atConnectionMetadata.isAuthenticated = false;
+      logger.severe('pkam authentication failed');
+      throw UnAuthenticatedException('pkam authentication failed');
+    }
+  }
 
-    //retrieve stored secret using sessionid and atsign
-    var storedSecret = await keyStore.get('private:$sessionID$atSign');
-    storedSecret = storedSecret?.data;
+  @visibleForTesting
+  Future<ApkamVerificationResult> handleApkamVerification(
+      String enrollId, String atSign) async {
+    String enrollmentKey =
+        '$enrollId.$newEnrollmentKeyPattern.$enrollManageNamespace$atSign';
+    var enrollData = await keyStore.get(enrollmentKey);
+    final atData = enrollData.data;
+    logger.finer('fetched enrollData: $atData');
```

Comment:
Please remove this log statement. Logging entire enrollment data which includes encrypted keys makes log noisy. Please remove any log statment's that contains the public/private key pair

Reply:
addressed as part of [270b30a](https://github.com/atsign-foundation/at_server/pull/1520/commits/270b30a26cf30cd073e04d917877ec274009985e)